doc: fix word wrapping for api stability boxes

saadq opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)
doc

---

This is meant to fix the issue that was opened [here](https://github.com/nodejs/nodejs.org/issues/1337) originally.

Fixes: https://github.com/nodejs/nodejs.org/issues/1337

Before: 
<img width="1146" alt="screen shot 2017-08-13 at 12 21 52 pm" src="https://user-images.githubusercontent.com/5678694/29251404-10390ecc-8022-11e7-80d5-177fad95096e.png">

After:
<img width="1144" alt="screen shot 2017-08-13 at 12 22 02 pm" src="https://user-images.githubusercontent.com/5678694/29251405-13973422-8022-11e7-8f6e-84e9c911914d.png">

silverwind commented:
Is there a reason why these boxes are `<pre>`? Doesn't seem semantically correct to me.

I'd change them to `<div>` and add `padding: 1em` to them.

refack commented:
> Is there a reason why these boxes are &lt;pre>? Doesn't seem semantically correct to me, and the fact that they inherit rules targeting code blocks isn't ideal either.

I think it's because it's rendered markdown. A simple style change might be to replace the ` ``` ` with `>`
<br>
![image](https://user-images.githubusercontent.com/96947/29252959-6f6c6af2-803f-11e7-8b85-202681404ebd.png)

or exclude `pre[lang=txt]` from the css

silverwind commented:
@refack nope, they are generated by our own code:

https://github.com/nodejs/node/blob/314217f8fbec156eff6dd61d71b3cd5327c41a5b/tools/doc/html.js#L403

refack commented:
Mostly enjoying the new code line quoting 👐 

> @refack nope, they are generated by our own code:

AFAICT that's the header for each API endpoint:
https://github.com/nodejs/node/blob/314217f8fbec156eff6dd61d71b3cd5327c41a5b/tools/doc/html.js#L33
https://github.com/nodejs/node/blob/314217f8fbec156eff6dd61d71b3cd5327c41a5b/tools/doc/html.js#L397-L406
That maps:
https://github.com/nodejs/node/blob/2e7ccc252e6c76835e892d135a29c45eb91ab5a4/doc/api/crypto.md#L678-L684
to:
![image](https://user-images.githubusercontent.com/96947/29253543-c2284e88-804d-11e7-95f7-789bca7e46f3.png)

---

Anyway let's do both: replace the markup and the rendering code

silverwind commented:
@refack I'm still pretty sure that `<pre>` comes from there, the changelog (`.api_metadata`) is generated elsewhere.

Anyways, my suggestion to @saadq would be to replace the `<pre>` with a `<div>` in `html.js` and add `padding: 1em` to `.api_stability` in CSS.

saadq commented:
I agree that `pre` probably wasn't that semantically correct, so I changed it to `div` as you suggested. However, this cause some stylistic issues:

**Before**
<img width="1146" alt="screen shot 2017-08-13 at 11 26 55 pm" src="https://user-images.githubusercontent.com/5678694/29257361-5f871516-807f-11e7-8e9f-908675b26816.png">

The `font-weight` became a lot bolder and the `line-height` became smaller, so I added a couple of styles to make it look like it did when it was still using `pre`.

**After**
<img width="1143" alt="screen shot 2017-08-13 at 11 31 02 pm" src="https://user-images.githubusercontent.com/5678694/29257381-8ae30c7e-807f-11e7-9324-2a1b29899d95.png">

Does this look alright to everyone?

TimothyGu commented on doc/api_assets/style.css:
```diff
@@ -93,7 +93,10 @@ em code {
+  white-space: pre-wrap;
+  word-wrap: break-word;
+  padding: 1em;
+  line-height: 1.5;
```
Are these (esp. the font and wrapping ones) still necessary now that the container has been switched to a `<div>`?

saadq replied:
Ahh, you're right. Since it's not `pre` anymore, the wrapping is no longer an issue. The `padding` and `line-height` are still necessary though.

saadq replied:
I reverted the wrapping changes. Just to clarify, I don't know if the `line-height` is strictly "necessary", I just added it because that was what the `pre` had originally. I don't know if you all think it looks fine without it.

Without `line-height`:
<img width="1143" alt="screen shot 2017-08-14 at 5 18 42 am" src="https://user-images.githubusercontent.com/5678694/29265727-1f1436ea-80b0-11e7-8990-0358673653d1.png">


With `line-height`:
<img width="1142" alt="screen shot 2017-08-14 at 5 18 53 am" src="https://user-images.githubusercontent.com/5678694/29265729-2311f944-80b0-11e7-8728-34f25e871cac.png">

silverwind replied:
I think I like it more with the padding. Another option would be to make it a `<p>` which already sets 1.5em line-height, so you save on declaration.

silverwind replied:
Thought, it might be unexpected to have `<p>` and those boxes share CSS. So, LGTM as-is.

tniessen commented:
Landed in eab2bea46fef091b8bf551eb9702b6da482048b3.
